src/devices/DCMotor/DCMotor.cs

```diff
@@ -20,9 +20,11 @@ public abstract class DCMotor : IDisposable
         /// Constructs generic <see cref="DCMotor"/> instance
         /// </summary>
         /// <param name="controller"><see cref="GpioController"/> related with operations on pins</param>
-        protected DCMotor(GpioController controller)
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        protected DCMotor(GpioController controller, bool shouldDispose = true)
```

Comment:
You don't really need to default shouldDispose here since this is a protected constructor and we are only calling it from our subclasses.

Reply:
Right, I'll adjust this one.